content/en/platform/corda/4.12/enterprise/node/operating/tvu/tvu-cli.md

```diff
@@ -0,0 +1,112 @@
+---
+date: '2023-12-15'
+section_menu: corda-enterprise-4-12
+menu:
+  corda-enterprise-4-12:
+    identifier: corda-enterprise-4-12-tvu-cli
+    parent: corda-enterprise-4-12-tvu
+tags:
+- tvu cli
+- tvu
+- transaction validator utility
+title: Transaction Validator Utility CLI parameters
+weight: 200
+---
+
+# Transaction Validator Utility CLI parameters
+
+The following section lists all the Transaction Validator Utility (TVU) CLI parameters. You can use them to perform various actions on transactions present in your database.
+
+### -b, --base-directory
+
+Absolute path to node's base directory for reading configuration directly from a node. If this option is not supplied, then the current directory is taken as the node's base directory.
+
+{{< note >}}
+If `-b` or `--base-directory` is specified, then CorDapps are also loaded from the supplied `base-directory/cordapps` directory and transaction deserialization is enabled by default.
+{{< /note >}}
+
+Example: `-b /corda/cordapp-template-java/build/nodes/PartyA`
+
+### -c, --class-load
+
+Full class name used for transaction processing. Class must implement `(SignedTransactionWithTimestamp) -> Unit` and should be placed under node's drivers directory: `/drivers`. The node's base-directory is taken as specified by the configuration in `-b` or `--base-directory` parameter.
+
+If this option is not provided, then the utility proceeds with default transaction validation which includes transaction verification and deserialization.
+
+Example: `-c net.corda.tvu.SampleApp`
+
+### --cordapp-dir
+
+Give absolute path to a directory containing CorDapps. You can provide this option zero or more times in the following way: `--cordapp-dir --cordappp-dir --cordapp-dir ...`. Provided path must be a directory. Specifying this parameter enables full transaction deserialization validation.
+
+### -d, --datasource
+
+Consists of all the properties required by [HikariCP](https://github.com/brettwooldridge/HikariCP) to establish a database connection. Check the required Hikari properties in the [Corda node configuration](({{< relref "../node-database-developer.html#2-corda-node-configuration" >}})) section.
```

Comment:
The URL given for HikariCP properties is a stale one from an older branch. The new and updated URL is given in 4.11 branch.
[Required Hikari Properties](https://docs.r3.com/en/platform/corda/4.10/enterprise/node/operating/node-database-developer.html#2-corda-node-configuration)